pylabrobot/resources/resource.py

```diff
@@ -200,10 +212,16 @@ def get_absolute_location(self, x: str = "l", y: str = "f", z: str = "b") -> Coo
     if self.parent is None:
       return self.location
     parent_pos = self.parent.get_absolute_location()
-    parent_rotation_matrix = self.parent.get_absolute_rotation().get_rotation_matrix()
-    local_vector = (self.location + self.get_anchor(x=x, y=y, z=z)).vector()
-    rotated_position = Coordinate(*matrix_vector_multiply_3x3(parent_rotation_matrix, local_vector))
-    return parent_pos + rotated_position
+
+    rotated_location = Coordinate(*matrix_vector_multiply_3x3(
+        self.parent.get_absolute_rotation().get_rotation_matrix(),
+        self.location.vector()
+    ))
+    rotated_anchor = Coordinate(*matrix_vector_multiply_3x3(
+        self.get_absolute_rotation().get_rotation_matrix(),
+        self.get_anchor(x=x, y=y, z=z).vector()
+    ))
+    return parent_pos + rotated_location + rotated_anchor
```

Comment:
location and anchor are actually in different coordinate systems, so they should be rotated differently